File: python/leetcode1.py
class Solution:
    def twoSum(self, nums, target):
        item_idx_map = {}
        for i in range(len(nums)):
            # 获取另一个数的下标，如果不存在返回None
            j = item_idx_map.get(target - nums[i], None)
            if j is not None:
                return [i, j]
            else:
                item_idx_map[nums[i]] = i
        return []

File: cpp/leetcode1.cpp
#include<vector>
#include<unordered_map>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;
        for (int i = 0; i < nums.size(); i++) {
            auto j = map.find(target - nums[i]);
            if (j != map.end()) {
                return {j->second, i};
            }
            else
                map[nums[i]] = i;
        }
        return {};
    }
};